[New Output] Support Note Output

ikelax opened the pull request:
Adresses #208.

### Changes

Add a capture group for the Note output.

### Testing

- Added new unit tests

ikelax committed:
Add support for the note output and tests for the parser and the renderers. Additionally, I had to decrease the magic numbers in ParsingTests.swift.

ikelax commented:
@cpisciotta Currently, the tests do not check if the strings are coloured correctly if I am not mistaken. Do you think this should be covered? What do you think about the formatting I chose for this output?

cpisciotta commented on Sources/XcbeautifyLib/CaptureGroups.swift:
```diff
@@ -1866,3 +1866,24 @@ struct SwiftDriverJobDiscoveryCompilingCaptureGroup: CaptureGroup {
+    /// Regular expression captured groups:
+    /// $1 = note
+    /// $2 = information
+    static let regex = Regex(pattern: "^(note:) (.*)")
```
nit: We can change this to `"^note: (.*)$".

No need to capture `note`, since it's a static keyword.

ikelax replied:
I will fix it.

cpisciotta reviewed (commented):
Hey @ikelax! I appreciate your contribution. Happy to get this merged in, but can you fix the conflicts first? Thanks!

cpisciotta commented:
> @cpisciotta Currently, the tests do not check if the strings are coloured correctly if I am not mistaken. Do you think this should be covered? What do you think about the formatting I chose for this output?

I think it LGTM! No need to test the colored output.

ikelax committed:
Merge branch 'main' into issue-208-support-note-output

ikelax committed:
Feedback from review and fix issues after merge

ikelax committed:
Remove comment

In the last commit, I removed the capturing of "note: " because it is a
static keyword and forgot to adapt the comment. Only the actual note is
captured.

ikelax committed:
Fix test

Two of the magic numbers in ParsingTests had decreased because more
lines are captured by the parser. So, I had to adjust it.

ikelax committed:
Add test for Azure DevOps pipeline renderer

ikelax commented:
@cpisciotta Thank you for review! Sorry, for the delay. I fixed the merge conflicts, added a test for the Azure DevOps pipeline renderer and made the change suggested by you.

cpisciotta commented on Tests/XcbeautifyLibTests/ParserTests.swift:
```diff
@@ -327,4 +327,39 @@ final class ParserTests: XCTestCase {
+        }
+    }
+
+    func testNotMatchNote() throws {
+        let inputs = [
+            "in the note middle",
+            "in the note: middle",
+            "note Building targets in dependency order",
+            "Note Metadata extraction skipped.",
+            "Target dependency graph (12 targets) note",
+            "Target dependency graph (12 targets) note:",
+            "Target dependency graph (12 targets): note:",
+        ]
+
+        for input in inputs {
+            XCTAssertNil(parser.parse(line: input))
+        }
+    }
```
Nice. Thanks for adding this test.

cpisciotta commented:
Thanks for your contribution @ikelax!
